src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -111,7 +111,8 @@ void DlgPrefWaveform::slotUpdate() {
     midVisualGain->setValue(factory->getVisualGain(WaveformWidgetFactory::Mid));
     highVisualGain->setValue(factory->getVisualGain(WaveformWidgetFactory::High));
     normalizeOverviewCheckBox->setChecked(factory->isOverviewNormalized());
-    defaultZoomComboBox->setCurrentIndex(factory->getDefaultZoom() - 1);
+    // Round zoom to int to get a default zoom index.
+    defaultZoomComboBox->setCurrentIndex(static_cast<int>(factory->getDefaultZoom()) - 1);
```

Comment:
Can this result in a situation where a user accidentally has their zoom level changed slightly by rounding because they press Apply in the preferences?

Reply:
The preferences only controls the default zoom level, so hitting apply only updates the saved value in the config.  It doesn't change the zoom level if it's been changed from the default.  Changing the selection in the preferences *does* update the zoom level to those values.   All in all the experience seems ok.

It used to be possible to right click on the waveform to reset to the default zoom level.  I don't know why that was removed, would it be ok to add it back in?  Right click-and-hold is used for fast track navigation so I guess that's a little overloaded.

Reply:
Okay, thanks for clarifying. Sorry for the delay reviewing this. I'm going over old PRs now that we finally got 2.2.0 released.

> It used to be possible to right click on the waveform to reset to the default zoom level. I don't know why that was removed, would it be ok to add it back in? Right click-and-hold is used for fast track navigation so I guess that's a little overloaded.

Seems reasonable to me. I didn't know about the right click and drag action for quick seeking. I don't really understand the use case considering it's much easier to seek where you want on the overview waveforms. If you want to do this, please open a new PR.